From b837f10265ebdde3ecd02fbf4baf0d141c4dfae2 Mon Sep 17 00:00:00 2001
From: Pranav Vashi <neobuddy89@gmail.com>
Date: Sun, 20 Oct 2024 16:46:12 +0530
Subject: [PATCH 5/6] install: Do not check ro.build.tags on user builds

Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
Signed-off-by: Dmitrii <bankersenator@gmail.com>
---
 install/install.cpp | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/install/install.cpp b/install/install.cpp
index ae4487f1..e16d505d 100644
--- a/install/install.cpp
+++ b/install/install.cpp
@@ -198,17 +198,6 @@ static bool CheckAbSpecificMetadata(const std::map<std::string, std::string>& me
       undeclared_downgrade = true;
     }
   }
-  const auto post_build = get_value(metadata, "post-build");
-  const auto build_fingerprint = android::base::Tokenize(post_build, "/");
-  if (!build_fingerprint.empty() && android::base::GetProperty("ro.build.type", "") == "user") {
-    const auto& post_build_tag = build_fingerprint.back();
-    const auto build_tag = android::base::GetProperty("ro.build.tags", "");
-    if (build_tag != post_build_tag) {
-      LOG(ERROR) << "Post build-tag " << post_build_tag << " does not match device build tag "
-                 << build_tag;
-      return false;
-    }
-  }
 
   if (!spl_downgrade_approved && undeclared_downgrade &&
       !(ui->IsTextVisible() && ask_to_continue_downgrade(ui->GetDevice()))) {
-- 
2.34.1

